### PR TITLE
OPHJOD-215: Add code coverage report to sonarcloud

### PR DESCRIPTION
**Pull request by henrikuo**
## Description

Enable SonarCloud code coverage report

## Related JIRA ticket

https://jira.eduuni.fi/browse/OPHJOD-215

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -65,6 +65,17 @@ jobs:
           path: playwright-report/
           retention-days: 30
 
+      # SonarCloud analysis
+      - name: SonarCloud
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # Shallow clones should be disabled for a better relevancy of analysis
```
Comment (jhyoty) [inline .github/workflows/build.yml:72]: Unnecessary to re-do the checkout -- move the fetch-depth:0 to the first checkout

File: vite.config.ts [hunk 2]
```diff
@@ -10,6 +10,10 @@ export default defineConfig({
   test: {
     environment: 'jsdom',
     exclude: [...configDefaults.exclude, 'e2e'],
+    coverage: {
+      provider: 'v8',
+      reporter: ['text', 'html', 'clover', 'json', 'lcov'],
```
Comment (jhyoty) [inline vite.config.ts:15]: maybe unnecessary to produce all the reports -- lcov (or lcovonly since the report is visualized in Sonar) could be sufficient

**Comment by sauanto** [comment]
The ticket number is missing from the last commit. Also commit title should start with the capital letter.

**Comment by henrikuo** [comment]
> The ticket number is missing from the last commit. Also commit title should start with the capital letter.

Fixed last commit message to contain ticket number and capital letters.

**Review by jhyoty — APPROVED** [review]

**Pull request merged**